src/io/fastq.rs

```diff
@@ -3,21 +3,25 @@
 // This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! FastQ reading and writing.
+//! FastQ reading.
 //!
 //! # Example
 //!
 //! ```
 //! use std::io;
 //! use bio::io::fastq;
-//! let reader = fastq::Reader::new(io::stdin());
-//! let records = reader.records().map(|r| r.unwrap());
-//! for record in records {
-//!     let check = record.check();
-//!     if check.is_err() {
-//!         panic!("I got a rubbish record!")
-//!     }
-//!     // your record is ok - do something with it...
+//! let mut reader = fastq::Reader::new(io::stdin());
+//! let mut record = fastq::Record::new();
+//! loop{
```

Comment:
It might be cleaner and clearer doing this in a `while` loop. That way you don't have to put in a `break` statement.
https://github.com/lh3/biofast/blob/96f518aeea90aa6b9f01e1103fa567bbd27beee5/fqcnt/fqcnt_rs1_rustbio.rs#L58-L70

Reply:
I was trying to avoid doing another record-reading before the loop. But I guess that one extra line is less confusing than the break statement.

Reply:
Have a look if you like the new while loop. And if it's correct... :D